File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    phone_book.sort() 
    num = 0
    for i in range(1, len(phone_book)):
        if phone_book[num] == phone_book[i][:len(phone_book[num])]:
            return False
        else:
            num = i
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.Arrays;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Arrays.sort(phone_book);
        for (int i=1; i<phone_book.length; i++) {
            if ((phone_book[i].length() >= phone_book[i-1].length()) &&
                (phone_book[i-1].equals(phone_book[i].substring(0, phone_book[i-1].length()))))
                return false;
        }
        
        return answer;
    }
}